packages/hoppscotch-data/src/rest/index.ts

```diff
@@ -1,66 +1,54 @@
-import cloneDeep from "lodash/cloneDeep"
+import { createVersionedEntity, InferredEntity } from "verzod";
 import * as Eq from "fp-ts/Eq"
 import * as S from "fp-ts/string"
-import { ValidContentTypes } from "./content-types"
-import { HoppRESTAuth } from "./HoppRESTAuth"
-import { lodashIsEqualEq, mapThenEq, undefinedEq } from "../utils/eq"
+import { V0_SCHEMA } from "./v/0"
+import V0_VERSION from "./v/0"
+import V1_VERSION from "./v/1"
+import { lodashIsEqualEq, mapThenEq, undefinedEq } from "../utils/eq";
+import cloneDeep from "lodash/cloneDeep"
+import { HoppRESTAuth, HoppRESTReqBody, HoppRESTHeaders, HoppRESTParams } from "./v/1"
 
 export * from "./content-types"
-export * from "./HoppRESTAuth"
-
-export const RESTReqSchemaVersion = "1"
-
-export type HoppRESTParam = {
-  key: string
-  value: string
-  active: boolean
-}
-
-export type HoppRESTHeader = {
-  key: string
-  value: string
-  active: boolean
-}
-
-export type FormDataKeyValue = {
-  key: string
-  active: boolean
-} & ({ isFile: true; value: Blob[] } | { isFile: false; value: string })
-
-export type HoppRESTReqBodyFormData = {
-  contentType: "multipart/form-data"
-  body: FormDataKeyValue[]
-}
-
-export type HoppRESTReqBody =
-  | {
-      contentType: Exclude<ValidContentTypes, "multipart/form-data">
-      body: string
-    }
-  | HoppRESTReqBodyFormData
-  | {
-      contentType: null
-      body: null
+export {
+  FormDataKeyValue,
+  HoppRESTReqBodyFormData,
+  HoppRESTAuth,
+  HoppRESTAuthAPIKey,
+  HoppRESTAuthBasic,
+  HoppRESTAuthBearer,
+  HoppRESTAuthNone,
+  HoppRESTAuthOAuth2,
+  HoppRESTReqBody
+} from "./v/1"
+
+export const HoppRESTRequest = createVersionedEntity({
+  latestVersion: 1,
+  versionMap: {
+    0: V0_VERSION,
+    1: V1_VERSION
+  },
+  getVersion(data) {
+    // For V1 onwards we have the v string storing the number
+    if (
+      typeof data === "object"
+      && data !== null
+      && "v" in data
+      && typeof data.v === "string"
+      && !Number.isNaN(parseInt(data.v))
+    ) {
+      return parseInt(data.v)
     }
 
-export interface HoppRESTRequest {
-  v: string
-  id?: string // Firebase Firestore ID
+    // For V0 we have to check the schema
+    const result = V0_SCHEMA.safeParse(data)
```

Comment:
```suggestion
    const result = V0_VERSION.schema.safeParse(data)
```

Can we refer to the schema from the default export?